backend/app/adapter/graphql/resolver/change.go

```diff
@@ -0,0 +1,35 @@
+package resolver
+
+import (
+	"github.com/short-d/short/app/adapter/graphql/scalar"
+	"github.com/short-d/short/app/entity"
+)
+
+// Change represents single change entity in ChangeLog
+type Change struct {
+	change entity.Change
+}
+
+// ID retrieves ID of Change entity
+func (c Change) ID() string {
+	return c.change.ID
+}
+
+// Title retrieves Title of Change entity
+func (c Change) Title() *string {
+	return &c.change.Title
+}
+
+// SummaryMarkdown retrieves SummaryMarkdown of Change entity
+func (c Change) SummaryMarkdown() *string {
+	return c.change.SummaryMarkdown
+}
+
+// ReleasedAt retrieves ReleasedAt of Change entity
+func (c Change) ReleasedAt() scalar.Time {
+	return scalar.Time{Time: *c.change.ReleasedAt}
```

Comment:
Are you sure we don't have here `nil pointer exception`?

Reply:
schema is this:
```
type Change {
  	id: String!
  	title: String
  	summaryMarkdown: String
  	releasedAt: Time!
}
```

entity is this:
```

type Change struct {
	ID              string
	Title           string
	SummaryMarkdown *string
	ReleasedAt      *time.Time
}
```

should i change entity to match the schema:
```
type Change struct {
	ID              string
	Title           *string
	SummaryMarkdown *string
	ReleasedAt      time.Time
}
```

Reply:
Also according to schema title can be nil, that should not happen right?

Reply:
@kapillamba4 
```golang
func (c Change) ReleasedAt() scalar.Time {
         if c.change.ReleasedAt == nil {
             return nil
         }
	return scalar.Time{Time: *c.change.ReleasedAt}
}
```
`*nil` will crash the service.

Reply:
@kapillamba4 Yes. `Title` can't be empty.

Reply:
this gives error "cannot use nil as type scalar.Time in return argument"

> @kapillamba4
> 
> ```go
> func (c Change) ReleasedAt() scalar.Time {
>          if c.change.ReleasedAt == nil {
>              return nil
>          }
> 	return scalar.Time{Time: *c.change.ReleasedAt}
> }
> ```
> 
> `*nil` will crash the service.

Reply:
You can change the function signature for this case:

```go
func (c Change) ReleasedAt() scalar.Time
```

As you are not sure that `c.change.ReleasedAt` is always a valid pointer

Reply:
then we would have to change the schema to:

```
type Change {
  	id: String!
  	title: String
  	summaryMarkdown: String
  	releasedAt: Time
}

```

Reply:
updated